File: Lokalano-partnerstvo/client/src/app/shared/models/dogadjaj.ts
export interface Dogadjaj {
    id: number;
    naziv: string;
    datumObjave: Date;
    datumPocetka: Date;
    vrijemePocetka: string;
    fokus: boolean;
    objavio: string;
    dogadjajKategorija: string;
    dogadjajKategorijaId: number;
    opis: string;
    imageUrl: string;
    photo: Photo;
  }

export interface Photo {
    id: number;
    imageUrl: string;
    fileName: string;
  }
export interface DogadjajToCreate {
    naziv: string;
    datumObjave: string;
    datumPocetka: string;
    vrijemePocetka: string;
    dogadjajKategorijaId: number;
    opis: string;
    objavio: string;
    imageUrl: string;
  }

export class DogadjajFormValues implements DogadjajToCreate {
    naziv = '';
    datumObjave = Date.now.toString();
    datumPocetka = Date.now.toString();
    vrijemePocetka = '';
    dogadjajKategorijaId: number;
    opis = '';
    objavio = '';
    imageUrl = '';

    constructor(init?: DogadjajFormValues) {
      Object.assign(this, init);
    }
  }

File: Lokalano-partnerstvo/Core/Entities/Dogadjaj.cs
using System;

namespace Core.Entities
{
    public class Dogadjaj : BaseEntity
    {
        public string Naziv { get; set; }
        public int DogadjajKategorijaId { get; set; }
        public DogadjajKategorija DogadjajKategorija { get; set; }
        public string Objavio { get; set; }
        public string Opis { get; set; }
        public DateTime DatumObjave { get; set; }
        public DateTime DatumPocetka { get; set; }
        public string VrijemePocetka { get; set; }
        public Photo Photo { get; set; }
        public int? PhotoId { get; set; }
        public string  ImageUrl { get; set; }


        public void AddPhoto(string imageUrl, string fileName)
        {
            var photo = new Photo
            {
                FileName = fileName,
                PictureUrl = imageUrl
            };

            Photo = photo;
        }
        public void RemovePhoto()
        {
            if (Photo != null)
            {
                Photo = null;
            }
        }
    }
}